test/TokenRequestTest.js

```diff
@@ -1,51 +1,304 @@
+      const actualTokenManager = await tokenRequest.tokenManager()
+      const actualVault = await tokenRequest.vault()
 
-        tokenManagerBase = await TokenManager.new()
-        MINT_ROLE = await tokenManagerBase.MINT_ROLE()
+      assert.strictEqual(actualTokenManager, tokenManager.address)
+      assert.strictEqual(actualVault, vault.address)
     })
+  })
 
+  describe('function setTokenManager(address _tokenManager)', () => {
     beforeEach(async () => {
-        await daoDeployment.deployBeforeEach(rootAccount)
-        const newTokenRequestAppReceipt = await daoDeployment.kernel
-            .newAppInstance('0x1234', tokenRequestBase.address, '0x', false, {from: rootAccount})
-        tokenRequest = await TokenRequest.at(deployedContract(newTokenRequestAppReceipt))
+      const newTokenManagerAppReceipt2 = await daoDeployment.kernel.newAppInstance(
+        '0x5555',
+        tokenManagerBase.address,
+        '0x',
+        false,
+        { from: rootAccount }
+      )
+      tokenManager2 = await TokenManager.at(deployedContract(newTokenManagerAppReceipt2))
+    })
+    it('sets a token manager', async () => {
+      await daoDeployment.acl.createPermission(accounts[1], tokenRequest.address, SET_TOKEN_MANAGER_ROLE, rootAccount)
+
+      await tokenRequest.setTokenManager(tokenManager2.address, { from: accounts[1] })
+
+      const actualTokenManager = await tokenRequest.tokenManager()
+      assert.strictEqual(actualTokenManager, tokenManager2.address)
+    })
+  })
+
+  describe('function setVault(address _vault)', () => {
+    beforeEach(async () => {
+      const newVaultAppReceipt2 = await daoDeployment.kernel.newAppInstance('0x8889', vaultBase.address, '0x', false, {
+        from: rootAccount,
+      })
+      vault2 = await Vault.at(deployedContract(newVaultAppReceipt2))
+    })
+    it('sets a vault', async () => {
+      await daoDeployment.acl.createPermission(accounts[1], tokenRequest.address, SET_VAULT_ROLE, rootAccount)
+
+      await tokenRequest.setVault(vault2.address, { from: accounts[1] })
 
-        const newTokenManagerAppReceipt = await daoDeployment.kernel
-            .newAppInstance('0x5678', tokenManagerBase.address, '0x', false, {from: rootAccount})
-        tokenManager = await TokenManager.at(deployedContract(newTokenManagerAppReceipt))
+      const actualVault = await tokenRequest.vault()
+      assert.strictEqual(actualVault, vault2.address)
     })
+  })
 
-    describe('initialize(address _tokenManager, address _vault)', () => {
+  describe('createTokenRequest(address _depositToken, uint256 _depositAmount, uint256 _requestAmount)', () => {
+    it('creates a new token request on exchange for Ether', async () => {
+      const expectedTRBalance = 2000
+      const expectedNextTokenRequestId = 1
 
-        beforeEach(async () => {
-            await tokenRequest.initialize(tokenManager.address, vault)
-        })
+      await tokenRequest.createTokenRequest(ZERO_ADDRESS, 2000, 1, {
```

Comment:
We could declare a constant ROOT_ETHER_AMOUNT and use it like you did in line 177.